smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputBodyMiddleware.kt

```diff
@@ -30,13 +30,16 @@ class OperationInputBodyMiddleware(
         val inputShapeName = MiddlewareShapeUtils.inputSymbol(symbolProvider, model, op).name
         val outputShapeName = MiddlewareShapeUtils.outputSymbol(symbolProvider, model, op).name
         val errorShapeName = MiddlewareShapeUtils.outputErrorSymbolName(op)
+        if (alwaysSendBody) {
+            writer.write("$operationStackName.${middlewareStep.stringValue()}.intercept(position: ${position.stringValue()}, middleware: \$N<$inputShapeName, $outputShapeName, $errorShapeName>(alwaysSendBody: true))", ClientRuntimeTypes.Middleware.SerializableBodyMiddleware)
+            return
+        }
         val hasHttpBody = MiddlewareShapeUtils.hasHttpBody(model, op)
-        if (hasHttpBody || alwaysSendBody) {
-
-            val alwaysSendBodyParam = if (alwaysSendBody) "alwaysSendBody: true" else ""
-            val middlewareTypeName = if (MiddlewareShapeUtils.bodyIsHttpPayload(model, op)) "${inputShapeName}BodyMiddleware()" else "${ClientRuntimeTypes.Middleware.SerializableBodyMiddleware.fullName}<$inputShapeName, $outputShapeName, $errorShapeName>($alwaysSendBodyParam)"
-
-            writer.write("$operationStackName.${middlewareStep.stringValue()}.intercept(position: ${position.stringValue()}, middleware: $middlewareTypeName)")
+        if (!hasHttpBody) return
+        if (MiddlewareShapeUtils.bodyIsHttpPayload(model, op)) {
+            writer.write("$operationStackName.${middlewareStep.stringValue()}.intercept(position: ${position.stringValue()}, middleware: ${inputShapeName}BodyMiddleware())")
+        } else {
+            writer.write("$operationStackName.${middlewareStep.stringValue()}.intercept(position: ${position.stringValue()}, middleware: \$N<$inputShapeName, $outputShapeName, $errorShapeName>())", ClientRuntimeTypes.Middleware.SerializableBodyMiddleware)
```

Comment:
looks really good -- up to you, but i think this might read better:
```
        if (alwaysSendBody) {
             ......
        } else if (MiddlewareShapeUtils.hasHttpBody(model, op)) {
            if (MiddlewareShapeUtils.bodyIsHttpPayload(model, op)) {
                    ......
            } else {
                      ......
            }
       }
```

You can leave it as you have it, but if we wanna avoid all the return statements, the above would be a way of doing it.

Reply:
Also, is there a place where we need to update in code which calls `bodyIsHttpPayload` inorder to generate the specific BodyMiddleware?

Reply:
updated code cuz I agree and updated generated body middleware file to use the same function now as well